connectors/embeddings-vector-database/src/test/java/io/camunda/connector/model/EmbeddingsVectorDBRequestValidationTest.java

```diff
@@ -0,0 +1,130 @@
+import io.camunda.connector.model.embedding.splitter.RecursiveDocumentSplitter;
+import io.camunda.connector.model.embedding.vector.store.AzureAiSearchVectorStore;
+import io.camunda.connector.model.embedding.vector.store.AzureAuthentication;
+import io.camunda.connector.model.embedding.vector.store.ElasticsearchVectorStore;
+import io.camunda.connector.model.embedding.vector.store.EmbeddingsVectorStore;
+import io.camunda.connector.model.operation.EmbedDocumentOperation;
+import io.camunda.connector.model.operation.EmbedDocumentSource;
+import io.camunda.connector.model.operation.VectorDatabaseConnectorOperation;
+import jakarta.validation.Validator;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.validation.ValidationAutoConfiguration;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+/**
+ * Tests for validating {@link EmbeddingsVectorDBRequest}. Note that only a small subset of possible
+ * valid and invalid configurations are tested here.
+ */
+@ExtendWith(SpringExtension.class)
+@Import(ValidationAutoConfiguration.class)
+class EmbeddingsVectorDBRequestValidationTest {
+
+  @Autowired private Validator validator;
+
+  @Test
+  void validationShouldSucceedWhenAllFieldsArePresent() {
+    var request =
+        new EmbeddingsVectorDBRequest(
+            sampleOperation(), sampleModelProvider(), sampleVectorStore());
+    assertThat(validator.validate(request)).isEmpty();
+  }
+
+  @Test
+  void validationShouldFailWhenOperationIsNull() {
+    var request = new EmbeddingsVectorDBRequest(null, sampleModelProvider(), sampleVectorStore());
+    var violations = validator.validate(request);
+    assertThat(violations).hasSize(1);
+  }
+
+  @Test
+  void validationShouldFailWhenModelProviderIsNull() {
+    var request = new EmbeddingsVectorDBRequest(sampleOperation(), null, sampleVectorStore());
+    var violations = validator.validate(request);
+    assertThat(violations).hasSize(1);
+  }
+
+  @Test
+  void validationShouldFailWhenVectorStoreIsNull() {
+    var request = new EmbeddingsVectorDBRequest(sampleOperation(), sampleModelProvider(), null);
+    var violations = validator.validate(request);
+    assertThat(violations).hasSize(1);
+  }
+
+  @Test
+  void validationShouldFailWhenOperationIsInvalid() {
+    var request =
+        new EmbeddingsVectorDBRequest(
+            new EmbedDocumentOperation(
+                EmbedDocumentSource.PlainText,
+                "document content",
+                List.of(),
+                new RecursiveDocumentSplitter(-1, 0)),
+            sampleModelProvider(),
+            sampleVectorStore());
+    var violations = validator.validate(request);
+    assertThat(violations).hasSize(2);
```

Comment:
💭 Maybe we could assert the actual field & error message (or parts of it), as it would make it a bit easier to understand which validation is triggered by the constellation when looking at this test class.

Reply:
done!